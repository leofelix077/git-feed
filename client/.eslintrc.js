module.exports = {
  extends: "@druideirabr",
  overrides: [
    {
      files: ["*.js"],
      rules: {
        "@typescript-eslint/explicit-function-return-type": "off",
      },
    },
  ],
};
