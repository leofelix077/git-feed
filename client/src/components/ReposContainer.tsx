import React, { useState, useEffect } from "react";
import { useGetReposQuery, Repository } from "../graphql/graphql";
import { SearchBar } from "./SearchBar";
import { makeStyles, Typography, Link, Grid, Button } from "@material-ui/core";
import PropTypes from "prop-types";
import RepoItem from "./RepoItem";
import { useTranslation } from "react-i18next";
import { GITHUB_OAUTH_URL } from "../constants";
import GitHubIcon from "@material-ui/icons/GitHub";
import { RootState } from "../redux/rootReducer";
import { useSelector, useDispatch } from "react-redux";
import { requestLogout, requestLogin } from "../redux/authState";

interface ReposContainerProps {
  queryStringProp?: string;
}

const useStyles = makeStyles((theme) => ({
  container: {
    flex: 1,
  },
  loadingText: {
    fontSize: 12,
    color: "white",
  },
  loginButton: {
    backgroundColor: "white",
    padding: theme.spacing(1),
    color: "black",
    alignItems: "center",
    display: "flex",
    flexDirection: "row",
  },
  loginText: {
    marginLeft: theme.spacing(1),
    color: "black",
  },
}));

const ReposContainer: React.FC<ReposContainerProps> = ({
  queryStringProp,
}): ReturnType<React.FC<ReposContainerProps>> => {
  const [queryString, setQueryString] = useState(queryStringProp || "");
  const classes = useStyles();
  const { t } = useTranslation("repoContainer");
  const dispatch = useDispatch();

  const isLoggedIn = useSelector(
    (state: RootState) => state.authState.isLoggedIn
  );
  const userInfo = useSelector((state: RootState) => state.authState.user);

  const { data, error, loading } = useGetReposQuery({
    variables: {
      repos: 10,
      queryString: `${
        userInfo ? `user:${userInfo.user.login}` : ""
      } ${queryString}`,
    },
    skip: !isLoggedIn,
    errorPolicy: "all",
  });

  useEffect(() => {
    const url = window.location.href;
    const hasCode = url.includes("?code=");
    if (hasCode) {
      const splitUrl = url.split("?code=");
      window.history.pushState({}, "", splitUrl[0]);
      dispatch(requestLogin(splitUrl[1].slice(0, 20)));
    }
  }, [dispatch]);

  const handleTextChange = (
    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ): any => {
    setQueryString(event.target.value);
  };

  return (
    <div className={classes.container} key="repo-container">
      <SearchBar handleChange={handleTextChange} value={queryString} />
      <Grid
        container
        alignItems="center"
        justify="center"
        style={{ display: "flex" }}
      >
        {isLoggedIn ? (
          <Button onClick={() => dispatch(requestLogout())}>Logout</Button>
        ) : (
          <Link
            href={GITHUB_OAUTH_URL}
            type="button"
            className={classes.loginButton}
          >
            <GitHubIcon />
            <Typography className={classes.loginText}>
              Login with GitHub
            </Typography>
          </Link>
        )}
      </Grid>
      {loading && !error && (
        <Typography className={classes.loadingText}>{t("loading")}</Typography>
      )}
      {error && (
        <Typography className={classes.loadingText}>{error.message}</Typography>
      )}
      {data &&
        data.search.edges &&
        data.search.edges.map(
          (
            repo: any & {
              node: Pick<
                Repository,
                | "name"
                | "url"
                | "description"
                | "forkCount"
                | "createdAt"
                | "pushedAt"
                | "updatedAt"
              >;
            }
          ) => <RepoItem key={repo.node.url} repo={repo.node} />
        )}
    </div>
  );
};

ReposContainer.propTypes = {
  queryStringProp: PropTypes.string,
};

ReposContainer.defaultProps = {
  queryStringProp: "",
};

export default ReposContainer;
