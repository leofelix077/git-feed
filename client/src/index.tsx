/* eslint-disable func-names */
/* eslint-disable no-extend-native */
import "moment/locale/pt-br";
import "moment/locale/en-gb";
import App from "./components/App";
import React from "react";
import ReactDOM from "react-dom";
import { unregister } from "./serviceWorker";
import { Provider as ReduxStoreProvider } from "react-redux";
import { applyMiddleware } from "redux";
import { reducer, initialState } from "./redux/rootReducer";
import { MuiThemeProvider, createMuiTheme } from "@material-ui/core/styles";
import { I18nextProvider } from "react-i18next";
import i18next from "./lib/i18next";
import createSagaMiddleware from "redux-saga";
import store from "./store";
import rootSaga from "./redux/rootSaga";
import { ConnectedRouter, routerMiddleware } from "connected-react-router";
import { createBrowserHistory } from "history";
import { ApolloProvider } from "@apollo/client";
import { apolloClient } from "./lib/apollo";

const theme = createMuiTheme({
  palette: {
    primary: {
      light: "#5dee48",
      main: "#00fa02",
      dark: "#008800",
      contrastText: "#fff",
    },
    secondary: {
      light: "#ff6434",
      main: "#00fa02",
      dark: "#008800",
      contrastText: "#ffffff",
    },
  },
  typography: {
    fontFamily: "'Roboto', sans-serif;",
    allVariants: {
      color: "#8D8D8D",
    },
  },
  overrides: {
    MuiPaper: {
      root: {
        backgroundColor: "#1a1a1a",
      },
    },
    MuiButton: {
      outlinedSecondary: {
        color: "#00fa02",
        borderColor: "#008800",
      },
    },
    MuiRadio: {
      colorSecondary: {
        color: "#008800",
      },
    },
  },
});

const history = createBrowserHistory();
const sagaMiddleware = createSagaMiddleware();
const middleware = routerMiddleware(history);

const reduxMiddleware = [middleware, sagaMiddleware];

store.set(reducer(history), initialState, applyMiddleware(...reduxMiddleware));

sagaMiddleware.run(rootSaga);

ReactDOM.render(
  <I18nextProvider i18n={i18next}>
    <ApolloProvider client={apolloClient}>
      <MuiThemeProvider theme={theme}>
        <ReduxStoreProvider store={store.get()}>
          <ConnectedRouter history={history}>
            <App />
          </ConnectedRouter>
        </ReduxStoreProvider>
      </MuiThemeProvider>
    </ApolloProvider>
  </I18nextProvider>,
  document.getElementById("root")
);

unregister();
