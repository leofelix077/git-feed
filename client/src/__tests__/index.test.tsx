import React from "react";
import { mount } from "enzyme";
import { describe, expect, it } from "@jest/globals";
import "jsdom-global/register";
import mockdate from "mockdate";
import { MockedProvider } from "@apollo/client/testing";
import { GetReposDocument } from "../graphql/graphql";
import ReposContainer from "../components/ReposContainer";
import { act } from "react-dom/test-utils";

describe("repos container", () => {
  it("should render component with children", async () => {
    mockdate.set(1593118700000);

    const apolloMocks: any = [
      {
        request: {
          query: GetReposDocument,
          variables: {
            queryString: "test",
            repos: 10,
          },
        },
        result: {
          data: {
            search: {
              repositoryCount: 4564611,
              edges: [
                {
                  node: {
                    name: "pytest",
                    url: "https://github.com/pytest-dev/pytest",
                    description:
                      "The pytest framework makes it easy to write small tests, yet scales to support complex functional testing",
                    createdAt: "2015-06-15T20:28:27Z",
                    pushedAt: "2020-08-03T18:10:42Z",
                    updatedAt: "2020-08-03T16:32:42Z",
                    __typename: "Repository",
                  },
                  __typename: "SearchResultItemEdge",
                },
                {
                  node: {
                    name: "uiautomator2",
                    url: "https://github.com/openatx/uiautomator2",
                    description: "Android Uiautomator2 Python Wrapper",
                    createdAt: "2017-09-17T12:20:42Z",
                    pushedAt: "2020-08-03T11:35:16Z",
                    updatedAt: "2020-08-03T11:08:54Z",
                    __typename: "Repository",
                  },
                  __typename: "SearchResultItemEdge",
                },
                {
                  node: {
                    name: "gremlins.js",
                    url: "https://github.com/marmelab/gremlins.js",
                    description:
                      "Monkey testing library for web apps and Node.js",
                    createdAt: "2013-10-21T08:47:12Z",
                    pushedAt: "2020-07-21T16:03:16Z",
                    updatedAt: "2020-08-01T08:42:02Z",
                    __typename: "Repository",
                  },
                  __typename: "SearchResultItemEdge",
                },
                {
                  node: {
                    name: "jest",
                    url: "https://github.com/facebook/jest",
                    description: "Delightful JavaScript Testing.",
                    createdAt: "2013-12-10T00:18:04Z",
                    pushedAt: "2020-08-03T16:36:57Z",
                    updatedAt: "2020-08-03T15:42:50Z",
                    __typename: "Repository",
                  },
                  __typename: "SearchResultItemEdge",
                },
                {
                  node: {
                    name: "java-testdata-generator",
                    url:
                      "https://github.com/binarywang/java-testdata-generator",
                    description:
                      "使用Java实现的各种测试数据的随机生成工具，包括身份证号码，银行卡号，姓名，手机号等",
                    createdAt: "2016-01-21T05:45:44Z",
                    pushedAt: "2020-07-02T03:53:42Z",
                    updatedAt: "2020-07-22T08:29:14Z",
                    __typename: "Repository",
                  },
                  __typename: "SearchResultItemEdge",
                },
                {
                  node: {
                    name: "tcpcopy",
                    url: "https://github.com/session-replay-tools/tcpcopy",
                    description:
                      "An online request replication tool, also a tcp stream replay tool, fit for real testing, performance testing, stability testing, stress testing, load testing, smoke testing, etc",
                    createdAt: "2011-10-14T08:15:19Z",
                    pushedAt: "2019-07-09T17:00:37Z",
                    updatedAt: "2020-08-02T17:48:42Z",
                    __typename: "Repository",
                  },
                  __typename: "SearchResultItemEdge",
                },
                {
                  node: {
                    name: "timecop",
                    url: "https://github.com/travisjeffery/timecop",
                    description:
                      'A gem providing "time travel", "time freezing", and "time acceleration" capabilities, making it simple to test time-dependent code. It provides a unified method to mock Time.now, Date.today, and DateTime.now in a single call.',
                    createdAt: "2008-11-07T20:37:52Z",
                    pushedAt: "2020-07-31T06:19:27Z",
                    updatedAt: "2020-08-03T03:29:37Z",
                    __typename: "Repository",
                  },
                  __typename: "SearchResultItemEdge",
                },
                {
                  node: {
                    name: "minitest",
                    url: "https://github.com/seattlerb/minitest",
                    description:
                      "minitest provides a complete suite of testing facilities supporting TDD, BDD, mocking, and benchmarking.",
                    createdAt: "2009-02-18T07:40:21Z",
                    pushedAt: "2020-05-18T07:21:54Z",
                    updatedAt: "2020-08-02T04:49:01Z",
                    __typename: "Repository",
                  },
                  __typename: "SearchResultItemEdge",
                },
                {
                  node: {
                    name: "stressTestPlatform",
                    url: "https://github.com/zyanycall/stressTestPlatform",
                    description: "基于Jmeter实现的在线压测和管理Jmx的平台",
                    createdAt: "2015-08-14T02:01:09Z",
                    pushedAt: "2020-07-10T02:06:55Z",
                    updatedAt: "2020-07-31T08:44:45Z",
                    __typename: "Repository",
                  },
                  __typename: "SearchResultItemEdge",
                },
                {
                  node: {
                    name: "redux-mock-store",
                    url: "https://github.com/reduxjs/redux-mock-store",
                    description:
                      "A mock store for testing Redux async action creators and middleware.",
                    createdAt: "2015-10-26T17:04:37Z",
                    pushedAt: "2020-04-08T09:52:46Z",
                    updatedAt: "2020-08-03T10:13:01Z",
                    __typename: "Repository",
                  },
                  __typename: "SearchResultItemEdge",
                },
              ],
              __typename: "SearchResultItemConnection",
            },
          },
        },
      },
    ];

    let wrapper: any;
    await act(async () => {
      wrapper = mount(
        (
          <MockedProvider mocks={apolloMocks}>
            <ReposContainer queryStringProp="test" />
          </MockedProvider>
        ) as any
      );
      wrapper.update();
      await new Promise((resolve) => setTimeout(resolve));
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
