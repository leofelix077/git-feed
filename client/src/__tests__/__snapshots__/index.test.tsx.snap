// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repos container should render component with children 1`] = `"<div class=\\"makeStyles-container-1\\"><div class=\\"makeStyles-searchFieldContainer-4\\"><div class=\\"MuiFormControl-root MuiTextField-root makeStyles-textFieldRoot-5 makeStyles-searchField-3\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\" for=\\"queryString\\" id=\\"queryString-label\\">Search</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-6 MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" id=\\"queryString\\" type=\\"search\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch\\" value=\\"test\\"></div></div></div><div class=\\"MuiGrid-root makeStyles-container-97 MuiGrid-container MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"https://github.com/pytest-dev/pytest\\" target=\\"_blank\\">pytest</a><h6 class=\\"MuiTypography-root makeStyles-description-99 MuiTypography-body1\\">The pytest framework makes it easy to write small tests, yet scales to support complex functional testing</h6></div><div class=\\"MuiGrid-root makeStyles-container-97 MuiGrid-container MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"https://github.com/openatx/uiautomator2\\" target=\\"_blank\\">uiautomator2</a><h6 class=\\"MuiTypography-root makeStyles-description-99 MuiTypography-body1\\">Android Uiautomator2 Python Wrapper</h6></div><div class=\\"MuiGrid-root makeStyles-container-97 MuiGrid-container MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"https://github.com/marmelab/gremlins.js\\" target=\\"_blank\\">gremlins.js</a><h6 class=\\"MuiTypography-root makeStyles-description-99 MuiTypography-body1\\">Monkey testing library for web apps and Node.js</h6></div><div class=\\"MuiGrid-root makeStyles-container-97 MuiGrid-container MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"https://github.com/facebook/jest\\" target=\\"_blank\\">jest</a><h6 class=\\"MuiTypography-root makeStyles-description-99 MuiTypography-body1\\">Delightful JavaScript Testing.</h6></div><div class=\\"MuiGrid-root makeStyles-container-97 MuiGrid-container MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"https://github.com/binarywang/java-testdata-generator\\" target=\\"_blank\\">java-testdata-generator</a><h6 class=\\"MuiTypography-root makeStyles-description-99 MuiTypography-body1\\">使用Java实现的各种测试数据的随机生成工具，包括身份证号码，银行卡号，姓名，手机号等</h6></div><div class=\\"MuiGrid-root makeStyles-container-97 MuiGrid-container MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"https://github.com/session-replay-tools/tcpcopy\\" target=\\"_blank\\">tcpcopy</a><h6 class=\\"MuiTypography-root makeStyles-description-99 MuiTypography-body1\\">An online request replication tool, also a tcp stream replay tool, fit for real testing, performance testing, stability testing, stress testing, load testing, smoke testing, etc</h6></div><div class=\\"MuiGrid-root makeStyles-container-97 MuiGrid-container MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"https://github.com/travisjeffery/timecop\\" target=\\"_blank\\">timecop</a><h6 class=\\"MuiTypography-root makeStyles-description-99 MuiTypography-body1\\">A gem providing \\"time travel\\", \\"time freezing\\", and \\"time acceleration\\" capabilities, making it simple to test time-dependent code. It provides a unified method to mock Time.now, Date.today, and DateTime.now in a single call.</h6></div><div class=\\"MuiGrid-root makeStyles-container-97 MuiGrid-container MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"https://github.com/seattlerb/minitest\\" target=\\"_blank\\">minitest</a><h6 class=\\"MuiTypography-root makeStyles-description-99 MuiTypography-body1\\">minitest provides a complete suite of testing facilities supporting TDD, BDD, mocking, and benchmarking.</h6></div><div class=\\"MuiGrid-root makeStyles-container-97 MuiGrid-container MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"https://github.com/zyanycall/stressTestPlatform\\" target=\\"_blank\\">stressTestPlatform</a><h6 class=\\"MuiTypography-root makeStyles-description-99 MuiTypography-body1\\">基于Jmeter实现的在线压测和管理Jmx的平台</h6></div><div class=\\"MuiGrid-root makeStyles-container-97 MuiGrid-container MuiGrid-item\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"https://github.com/reduxjs/redux-mock-store\\" target=\\"_blank\\">redux-mock-store</a><h6 class=\\"MuiTypography-root makeStyles-description-99 MuiTypography-body1\\">A mock store for testing Redux async action creators and middleware.</h6></div></div>"`;
